bundles/org.openhab.binding.airq/src/main/java/org/openhab/binding/airq/internal/AirqHandler.java

```diff
@@ -0,0 +1,956 @@
+        } catch (Exception e) {
+            if (System.out != null) {
+                System.out.println("air-Q - airqHandler - decrypt(): Error while decrypting: " + e.toString());
+            }
+            return null;
+        }
+        return content;
+    }
+
+    public String encrypt(byte[] toencode, String password) {
+        String content = "";
+        logger.trace("air-Q - airqHandler - encrypt(): text to encode: {}", new String(toencode));
+        byte[] passkey = Arrays.copyOf(password.getBytes(), 32);
+        if (password.length() < 32) {
+            Arrays.fill(passkey, password.length(), 32, (byte) '0');
+        }
+        byte[] IV = new byte[16];
+        SecureRandom random = new SecureRandom();
+        random.nextBytes(IV);
+        SecretKey seckey = new SecretKeySpec(passkey, 0, passkey.length, "AES");
+        try {
+            Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
+            SecretKeySpec keySpec = new SecretKeySpec(seckey.getEncoded(), "AES");
+            IvParameterSpec ivSpec = new IvParameterSpec(IV);
+            cipher.init(Cipher.ENCRYPT_MODE, keySpec, ivSpec);
+            byte[] encryptedText = cipher.doFinal(toencode);
+            byte[] totaltext = new byte[16 + encryptedText.length];
+            System.arraycopy(IV, 0, totaltext, 0, 16);
+            System.arraycopy(encryptedText, 0, totaltext, 16, encryptedText.length);
+            byte[] encodedcontent = Base64.getEncoder().encode(totaltext);
+            logger.trace("air-Q - airqHandler - encrypt(): encrypted text: {}", encodedcontent);
+            content = new String(encodedcontent);
+        } catch (Exception e) {
+            if (System.out != null) {
+                System.out.println("air-Q - airqHandler - encrypt(): Error while encrypting: " + e.toString());
+            }
+        }
+        return content;
+    }
+
+    protected @Nullable String getDecryptedContentString(String url, String requestMethod, @Nullable String body) {
+        Result res = null;
+        String jsonAnswer = null;
+        res = getData(url, "GET", null);
+        if (res != null) {
+            String jsontext = res.getBody();
+            logger.trace("air-Q - airqHandler - getDecryptedContentString(): Result from doNetwork is {} with body={}",
+                    res, res.getBody());
+            // Gson code based on https://riptutorial.com/de/gson
+            Gson gson = new Gson();
+            JsonElement ans = gson.fromJson(jsontext, JsonElement.class);
+            if (ans != null) {
+                JsonObject jsonObj = ans.getAsJsonObject();
+                jsonAnswer = decrypt(jsonObj.get("content").getAsString().getBytes(), config.password);
+                if (jsonAnswer == null) {
+                    logger.warn("The air-Q data could not be decrypted. Probably the password is wrong.");
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Wrong password");
```

Comment:
Can you do a find on 'updateStatus(' and if you have a logger and updateStatus( together can you work out if logger can be removed as it is redundant. See the developer guidelines for logging.